stock_picking_consolidation_priority/models/stock_move.py

```diff
@@ -121,4 +121,19 @@ def _consolidate_priority(self):
                 [[("id", "in", move_ids)], self._consolidate_priority_domain()]
             )
         )
-        moves.picking_id.write(self._consolidate_priority_values())
+        for move in moves:
+            # Flag the move as urgent.
+            move.write(self._consolidate_priority_values())
+            # Flag the picking as urgent if it only contains urgent moves.
+            picking = move.picking_id
+            picking_moves = picking.move_lines.filtered(
+                lambda m: m.state not in ("cancel", "done")
+            )
+            if (
+                all(
+                    m.priority == self._consolidate_priority_value
+                    for m in picking_moves
+                )
+                and picking.priority is not self._consolidate_priority_value
+            ):
+                picking.write(self._consolidate_priority_values())
```

Comment:
This looks overkill to me.
Can't we factorize the calls to _consolidate_priority_values?
If there's multiple moves, each with the same picking_id, we might end up to prioritize it multiple times.
Is that ok?
Even though I don't have the full picture, I don't like this, and I think we can do better.

Reply:
Indeed. Something like:
```
        for move in moves:
            # Flag the move as urgent.
            move.write(self._consolidate_priority_values())
        for picking in moves.picking_id:
            # Flag the picking as urgent if it only contains urgent moves.
```

Reply:
We really wouldn't prioritize the same picking multiple times due to the `picking.priority is not self._consolidate_priority_value` condition, we would simply skip that picking.
I did it this way to avoid having to implement a second loop on pickings, and instead reuse the first one that we're doing on the moves.

Reply:
On each move, you loop on each move of the move's picking with filtered. So it is O(n²).
It's better to change :)

Reply:
Don't we still have to apply that filtered anyway to exclude cancelled and done moves?

Reply:
yes you do, but only once per picking, not once per move

Reply:
I don't see much of a difference in that case, we end up having to do loops everywhere anyway, but ok, updated. Thanks!